docs/development/dev-container-setup.md

```diff
@@ -0,0 +1,170 @@
+
+- **Go**: The Go programming language environment.
+- **Docker-in-Docker**: Allows running Docker commands inside the container.
+- **AWS CLI**: For interacting with AWS services.
+- **kubectl**, **Helm**: For Kubernetes development and testing.
+- **Starts Minikube**: Initializes a local Kubernetes cluster.
+- **Logs into GHCR**: Authenticates with GitHub Container Registry.
+- **Prepares Directories**: Sets up logs and data directories for S3 and IAM services, `.github/e2e_tests`.
+- **Deploys S3 and IAM Services**: Uses Docker Compose to start the required backend services.
+- **Port Forwarding**: Ports `8000` and `8600` are forwarded for accessing services running inside the container.
+- **Installs Testing Tools**: Ginkgo and Gomega for Go testing.
+- **Cleans Up Docker Images**: Prunes unused Docker images to save space.
+
+---
+
+## Opening the Dev Container
+
+### Use GitHub UI
+
+![Create Codespaces from GitHub UI](../images/create-codespaces-from-github-ui.png)
+
+### Open the Project in VS Code
+
+1. Open Visual Studio Code.
+2. Use the command palette (`Ctrl+Shift+P` or `Cmd+Shift+P`) and select **Remote-Containers: Open Folder in Container...**.
+3. Navigate to and select the project folder (`cosi-driver/`).
+
+VS Code will automatically detect the `.devcontainer/devcontainer.json` configuration and build the Dev Container. This may take a few minutes on the first run.
+
+---
+
+## Using the Dev Container
+
+Once the Dev Container is up and running, you will have access to the pre-configured development environment.
+
+There are 3 main scripts run in the CI:
+
+- **.github/scripts/setup_cosi_resources.sh**: setups up COSI CRDs, COSI controller, COSI sidecar, and Scality COSI driver.
+- **.github/scripts/e2e_test_bucket_creation.sh**: Sets up BucketClass, Bucket Claim, s3 secret(changed needed below), in kubernetes and tests create bucket using AWS CLI.
+- **.github/scripts/cleanup_cosi_resources.sh**: Deletes all researources created in above two steps.
+
+### Running Setup and Test Scripts
+
+In the integrated terminal inside VS Code, run the following commands to set up resources, run end-to-end tests, and capture logs:
+
+1. **Update endpoint in [secrets file](../../cosi-examples/s3-secret-for-cosi.yaml)**
+
+Change `COSI_S3_ENDPOINT` to the IP (`hostname -I | awk '{print $1}'`) of codespace instance.
+
+#### Example
+
+```bash
+$ hostname -I | awk '{print $1}'
+10.0.1.236
+```
+
+New secret file with updated `COSI_S3_ENDPOINT`
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: s3-secret-for-cosi
+  namespace: default
+type: Opaque
+stringData:
+  COSI_S3_ACCESS_KEY_ID: PBUOB68AVF39EVVAFNFL  # Plain text access key, generated in the CI
+  COSI_S3_SECRET_ACCESS_KEY: P+PK+uMB9spUc21huaQoOexqdJoV00tSnl+pc7t7  # Plain text secret key
+  COSI_S3_ENDPOINT: http://10.0.1.236:8000  # Plain text endpoint
+  COSI_S3_REGION: us-west-1  # Plain text region
+```
```

Comment:
format: I think you could add indentation to everything under 1 as you do for 2 and 3

Reply:
I removed this part as its automated.